Pre-release prep. Chapel 1.16.0: post-release version markings

awallace-cray opened the pull request:
THIS IS NOT CHAPEL RELEASE 1.16.0

This change prepares the source on master branch for the upcoming
release, currently scheduled for 5 Oct 2017.

This bumps the remaining release version numbers to 1.16.0, and
removes the string "pre-release" whereever it was added in
PR #5992.
    
The version numbers changed here were deliberately skipped when
other existing version numbers were bumped to 1.16 in PR #5992.
These version numbers were left at "1.15" to protect some
online docs generated from master from showing broken
hyperlinks or written instructions (ie, because URLs for
release 1.16.0 are not published yet).

bradcray commented:
High-level comment:  When we're ready for this change, I'd be inclined to merge it to master and then cut the branch rather than cutting the branch and then applying it there.  Rationale: (1) we'll need to update some of these version numbers (like the ones in the docs) on master anyway; (2) it gives us the chance to do some of our normal nightly testing on the release candidate as represented by master; (3) it keeps our major work on master, such that when future devs go looking for "How did we do that?" they'll find it on master rather than needing to go look at the branch.

bradcray commented on compiler/main/version.cpp:
```diff
@@ -31,7 +31,7 @@ void
   v += sprintf(v, "%d.%s.%s", MAJOR_VERSION, MINOR_VERSION, UPDATE_VERSION);
   if (strcmp(BUILD_VERSION, "0") != 0 || developer)
-    sprintf(v, " pre-release (%s)", BUILD_VERSION);
+    sprintf(v, ".%s", BUILD_VERSION);   // post-release
```
I'm a little disappointed to find that when I implemented this, I didn't make the printing or the pre-release-specific information configurable by flipping a single PRE_RELEASE #define or somesuch...  I thought I'd made the two formats coexist more than it seems I did...  Good intentions...

awallace-cray commented:
Remove "pre-release" from master, create the release branch, then put "pre-release" back in master?

bradcray commented on test/classes/initializers/generics/initAtomicsViaWrite.bad:
```diff
@@ -1,4 +1,4 @@
-initAtomicsViaWrite.chpl:3: internal error: GEN0433 chpl Version 1.16.0 pre-release (c0bf20b)
```
I'd be inclined to leave the .bad files as they were assuming that the script which is papering over SHAs and release numbers works for them (and I'd adjust the script if it doesn't).  Reason being that we should try and minimize our maintenance workload.  Also it'll be nice for the 1.17 release to be able to look at this PR and see which n files need to change at a minimum, and I worry we'd get into the practice of making updates like this needlessly.

bradcray commented:
Other than the comment on the .bad files, this looks good to me.  I didn't take any steps along the lines of (a) testing that things work; (b) looking for other version numbers that might've been overlooked, but could do so later this week.

awallace-cray commented:
@bradcray - Thanks. I will revert the ".bad" files. IMO, you don't need to test more things or look for more places with versions. I'll do that.

bradcray commented:
>  Remove "pre-release" from master, create the release branch, then put "pre-release" back in master?

Yeah, though I imagine there potentially being a day or two (or several) between those steps during which we're living with the release candidate on master:

* remove pre-release from master and bump some version numbers (the ones in READMEs)
* live in this state for some number of days to get nightly testing on it
* cut release branch
* restore pre-release on master and bump the other version numbers (the other ones)

As I type that, I'm having trouble remembering offhand why we didn't bump all the release numbers to 1.16 when we switched to this pre-release tagging (notably, the ones in QUICKSTART and chplenv).  If we were able to unify them all, this could become:

* remove pre-release from master
* live in this state for some number of days to get nightly testing on it
* cut release branch
* bump the version numbers and restore pre-release label

awallace-cray commented:
* Rebased to master as of 9/19 nightly builds.  
* Dropped the commit w removed "pre-release" from test .bad files. 
* Changed commentary according to new intention to merge this to master, then follow later with other PR(s) to bump versions to 1.17.0 and restore the "pre-release" markings in master.

awallace-cray commented:
@bradcray Please approve this PR to indicate when to merge it to master. Thanks!

awallace-cray commented:
Rebased to master as of 9/22 nightly builds.

bradcray commented:
I think this could go in anytime now that testing has settled down a bit and code freeze approaches.
